### ISSUE TITLE
va-modal - remove `ariaHiddenNodeExceptions` prop

### ISSUE THREAD

**Issue by RyanMunsch** [department-of-veterans-affairs/vets-design-system-documentation#5069]
## Description
Remove the `ariaHiddenNodeExceptions` prop and related logic from `va-modal`.

## Details

The `ariaHiddenNodeExceptions` prop allows elements outside of the modal to be focusable. This is considered bad practice with modals and violates [USWDS Guidance](https://designsystem.digital.gov/components/modal/#accessibility-guidance):

> Ensure keyboard focus is trapped inside the modal. When using a keyboard to test the modal, ensure focus does not move to the page behind it.

The prop is currently being used in [one component on `vets-website`](https://github.com/department-of-veterans-affairs/vets-website/blob/main/src/platform/site-wide/announcements/components/HomepageRedesignModal.jsx#L31), but that component is **not** being used in production as confirmed by the team's accessibility specialist (Slack record [here](https://wilcore.slack.com/archives/C08N8Q5E9K5/p1761253378804879?thread_ts=1761241062.658469&cid=C08N8Q5E9K5)).

## Acceptance criteria

- [ ] `ariaHiddenNodeExceptions` prop is removed
- [ ] Focus trap acts as expected - focus cannot leave the modal while the component instance is open.
